test/components/date-picker.spec.tsx

```diff
@@ -613,6 +613,106 @@ describe('The DatePicker Component', () => {
             });
         });
 
+        it('when in the month-view, clicking on the year should bring up the year-view'
+            + ' and the header date should show a decade', async () => {
+            const {driver: datePicker, waitForDom} = clientRenderer.render(
+                <DatePicker showDropdownOnInit value={JANUARY_FIRST}/>
+            ).withDriver(DatePickerTestDriver);
+
+            datePicker.clickOnHeader();
+
+            await waitForDom(() => {
+                expect(datePicker.monthView).to.be.present();
+                expect(datePicker.headerDate).to.have.text('2017');
+            });
+
+            datePicker.clickOnHeader();
+
+            await waitForDom(() => {
+                expect(datePicker.monthView).to.be.absent();
+                expect(datePicker.yearView).to.be.present();
+                expect(datePicker.headerDate).to.have.text('2012-2021');
+            });
+        });
+
+        it('when in the month-view, clicking on the arrows should not change the actual date', async () => {
```

Comment:
is the outcome of pressing the arrows in month view being tested anywhere? if not this is a good place for it, and during making sure the data value stays the same